cobbler/templar.py

```diff
@@ -139,9 +140,12 @@ def render(self, data_input, search_table, out_path, subject=None, template_type
             repstr = server
         search_table["http_server"] = repstr
 
-        for x in list(search_table.keys()):
-            if type(x) == str:
-                data_out = data_out.replace("@@%s@@" % str(x), str(search_table[str(x)]))
+        # string replacements for @@xyz@@ in data_out with prior regex lookups of keys
+        regex = r"@@[a-zA-Z]*@@"
```

Comment:
I think it is possible to include also stuff like numbers etc. in the variable names. If we don't want to support this I am fine with it but we would need to document this regression in the docs!

Reply:
That could very well be, but I'm not sure about it. I haven seen a key with a number yet.

Reply:
Regex is more greedy now. I've switched to `r"@@[\S]*@@"`.